program-runtime/src/invoke_context.rs

```diff
@@ -170,7 +217,15 @@ impl<'a> InvokeContext<'a> {
         feature_set: Arc<FeatureSet>,
         blockhash: Hash,
         lamports_per_signature: u64,
+        current_accounts_data_len: u64,
     ) -> Self {
+        let accounts_data_meter = {
+            assert!(current_accounts_data_len <= MAX_ACCOUNTS_DATA_LEN);
```

Comment:
Assert?

Reply:
Yes, figured it would be a good invariant to ensure. Is this not a good idea?

Reply:
The problem is that `assert!` can panic in production, which is bad. Use a `debug_assert!` instead.

Reply:
In this case use a better way to protect against errant cases that allows the cluster to continue and handle gracefully

Reply:
Done! Assert changed to debug_assert.